tests/end_to_end/helpers.h

```diff
@@ -235,5 +235,6 @@ typedef class _program_info_provider
     ebpf_extension_provider_t* provider;
 } program_info_provider_t;
 
+#define ETHERNET_TYPE_IPV4 0x0800
 std::vector<uint8_t>
-prepare_udp_packet(uint16_t udp_length);
\ No newline at end of file
+prepare_udp_packet(uint16_t udp_length, uint16_t eth_type = ETHERNET_TYPE_IPV4);
```

Comment:
```suggestion
prepare_udp_packet(uint16_t udp_length, uint16_t ethertype);
```
don't default to IPv4

Reply:
The reason why I did this is several other tests were using this helper function as well which were all IPv4. I did not want to churn those tests. But now I have made those changes.